src/cogent3/core/new_genetic_code.py

```diff
@@ -0,0 +1,438 @@
+"""Translates RNA or DNA string to amino acid sequence.
+
+Notes
+ ----
+
+* is used to denote termination (as per NCBI standard).
+
+Although the genetic code objects convert DNA to RNA and vice
+versa, lists of codons that they produce will be provided in DNA format.
+"""
+
+import collections
+import contextlib
+import dataclasses
+import itertools
+import typing
+
+import numpy
+
+from cogent3.core import new_alphabet, new_moltype
+from cogent3.util.table import Table
+
+
+OptStr = typing.Optional[str]
+SetStr = typing.Set[str]
+ConverterType = typing.Callable[[bytes, bytes], bytes]
+StrORInt = typing.Union[str, int]
+
+DNA = new_moltype.get_moltype("dna")
+RNA = new_moltype.get_moltype("rna")
```

Comment:
Can't place if these are used in the file? e.g. could be used on L95 `moltype: new_moltype.MolType = DNA`, but having the reference to the `new_*` module in-line is clearer.

Reply:
good point